src/main/scala/org/neo4j/spark/service/SchemaService.scala

```diff
@@ -0,0 +1,140 @@
+package org.neo4j.spark.service
+
+import java.util.Collections
+
+import org.apache.spark.sql.types.{DataType, DataTypes, StructField, StructType}
+import org.neo4j.cypherdsl.core.Cypher
+import org.neo4j.cypherdsl.core.renderer.Renderer
+import org.neo4j.driver.exceptions.ClientException
+import org.neo4j.driver.Session
+import org.neo4j.spark.util.Neo4jUtil
+import org.neo4j.spark.{DriverCache, Neo4jOptions, Neo4jQuery}
+
+import collection.JavaConverters._
+import scala.collection.mutable
+
+class SchemaService(private val options: Neo4jOptions, private val jobId: String) extends AutoCloseable {
+
+  import SchemaService._
+
+  import org.neo4j.spark.QueryType._
+
+  private val cypherRenderer = Renderer.getDefaultRenderer
+
+  private val driverCache: DriverCache = new DriverCache(options.connection, jobId)
+
+  private val session: Session = driverCache.getOrCreate().session(options.session.toNeo4jSession)
+
+  private def cypherToSparkType(cypherType: String): DataType = {
+    cypherType match {
+      case "Boolean" => DataTypes.BooleanType
+      case "String" => DataTypes.StringType
+      case "Long" => DataTypes.IntegerType
+      case "Double" => DataTypes.DoubleType
+      case "Point" | "InternalPoint2D" | "InternalPoint3D" => pointType
```

Comment:
let's talk about this more on friday.  Type mapping is a critical piece.  No change needed -- but want to understand options here.  In particular neo4j points are a bit strange (having x, y, z, and scheme) and I'm not sure about the spark side of this.